R/ard_stack.R

```diff
@@ -0,0 +1,130 @@
+#' Stack ARDs
+#'
+#' Stack multiple ARD calls sharing common input `data` and `by` variables.
+#' Optionally incorporate additional information on represented variables (i.e.
+#' big N's, missingness, attributes) and/or tidy for use in displays with
+#' `shuffle_ard()`.
+#'
+#' @param data (`data.frame`)\cr
+#'   a data frame
+#' @param by ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   columns to tabulate by in the series of ARD function calls
+#' @param ... ([`dynamic-dots`][dyn-dots])\cr
+#'   Series of ARD function calls to be run and stacked
+#' @param .compute_by_ns (`logical`)\cr
+#'   logical indicating whether to compute big N's for the `by` variable
+#'   Default is `TRUE`.
+#' @param .compute_missing (`logical`)\cr
+#'   logical indicating whether to include the results of `ard_missing()` for all
+#'   variables represented in the ARD. Default is `FALSE`.
+#' @param .add_attributes (`logical`)\cr
+#'   logical indicating whether to include the results of `ard_attributes()` for all
+#'   variables represented in the ARD. Default is `FALSE`.
+#' @param .shuffle (`logical`)\cr
+#'   logical indicating whether to perform `shuffle_ard()` on the final result.
+#'   Default is `FALSE`.
+#' @param .trim (`logical`)\cr
+#'   logical indicating whether to set `trim = TRUE` in the call to `shuffle_ard()`.
+#'   Default is `TRUE`. Will be ignored if `.shuffle = FALSE`.
+#'
+#' @return  a transformed ARD data frame (of class 'card' if `.shuffle = FALSE`)
+#'
+#' @export
+#' @examples
+#'
+#' ard_stack(
+#'   data = ADSL,
+#'   by = "ARM",
+#'   ard_categorical(variables = "AGEGR1"),
+#'   ard_continuous(variables = "AGE")
+#' )
+#'
+#' ard_stack(
+#'   data = ADSL,
+#'   by = "ARM",
+#'   ard_categorical(variables = "AGEGR1"),
+#'   ard_continuous(variables = "AGE"),
+#'   .shuffle = TRUE
+#' )
+#'
+ard_stack <- function(data,
+                      by = NULL,
+                      ...,
+                      .compute_by_ns = TRUE,
+                      .compute_missing = FALSE,
+                      .add_attributes = FALSE,
+                      .shuffle = FALSE,
+                      .trim = TRUE) {
```

Comment:
I'm only a little hesitant here because the trim filters to numeric results only. But I am all for keeping it simple especially in these stages. Will remove the arg.